client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-how-to-migrate/index.tsx

```diff
@@ -0,0 +1,121 @@
+import { Card } from '@automattic/components';
+import { StepContainer, SubTitle, Title } from '@automattic/onboarding';
+import { getQueryArg } from '@wordpress/url';
+import { useTranslate } from 'i18n-calypso';
+import DocumentHead from 'calypso/components/data/document-head';
+import { useAnalyzeUrlQuery } from 'calypso/data/site-profiler/use-analyze-url-query';
+import { useHostingProviderQuery } from 'calypso/data/site-profiler/use-hosting-provider-query';
+import { useSite } from 'calypso/landing/stepper/hooks/use-site';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { usePresalesChat } from 'calypso/lib/presales-chat';
+import useHostingProviderName from 'calypso/site-profiler/hooks/use-hosting-provider-name';
+import type { Step } from '../../types';
+
+import './style.scss';
+
+const SiteMigrationHowToMigrate: Step = function ( { navigation } ) {
+	const translate = useTranslate();
+	const site = useSite();
+	const importSiteQueryParam = getQueryArg( window.location.href, 'from' )?.toString() || '';
+
+	const options = [
+		{
+			label: translate( 'Do it for me' ),
+			description: translate(
+				"Share your site with us, and we'll review it and handle the migration if possible."
+			),
+			value: 'difm',
+			selected: true,
+		},
+		{
+			label: translate( "I'll do it myself" ),
+			description: translate(
+				'Install the plugin yourself, find the migration key and migrate the site.'
+			),
+			value: 'myself',
+		},
+	];
+
+	let importSiteHostName = '';
+
+	try {
+		importSiteHostName = new URL( importSiteQueryParam )?.hostname;
```

Comment:
Why do we need this try-catch?

Reply:
`new URL` throws an Error if we don't pass a valid URL to the constructor, and, in this case, we want it to fail silently.